src/overlays/actors/ovl_Obj_Boat/z_obj_boat.c

```diff
@@ -28,31 +32,164 @@ const ActorInit Obj_Boat_InitVars = {
+        if (this->unk15F != 0) {
+            this->unk15F--;
+        }
+    }
+    if (this->unk15C != this->unk15E) {
+        sp3A = func_80B9AF50(this, &sp2C);
+        if (Math_Vec3f_DistXZ(&this->dyna.actor.world.pos, &sp2C) < 200.0f) {
+            this->unk15C += this->unk15D;
+            if (this->unk15C == this->unk15E) {
+                if ((this->dyna.actor.params & 0x4000)) {
+                    this->unk15C = 0;
+                } else if (this->dyna.actor.speedXZ == 0.0f) {
+                    this->unk15C = 0;
+                    this->unk15D = -1;
+                }
+            }
+        } else {
+            sp3C = this->unk15D * ((this->dyna.actor.params & 0x4000) ? 5.0f : 3.0f);
+        }
+    }
+    if (player->csMode != 0x1A) {
+        Math_ScaledStepToS(&this->dyna.actor.shape.rot.y, sp3A, (s16)(s32)(fabsf(this->dyna.actor.speedXZ) * 40.0f));
+        this->dyna.actor.world.rot.y = this->dyna.actor.shape.rot.y;
+        Math_StepToF(&this->dyna.actor.speedXZ, sp3C, 0.05f);
+        Actor_MoveWithGravity(&this->dyna.actor);
+        if (this->dyna.actor.speedXZ != 0.0f) {
+            func_800B9010(&this->dyna.actor, 0x2154U);
+        }
+    }
+    func_80B9B124(this);
+}
+
+void func_80B9B428(ObjBoat* this, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    s32 pad;
+
+    if (Cutscene_CheckActorAction(globalCtx, 0x1FFU) != 0) {
+        CsCmdActorAction* var_a3 = globalCtx->csCtx.actorActions[Cutscene_GetActorActionIndex(globalCtx, 0x1FFU)];
+        if (this->unk15F != var_a3->action) {
+            this->dyna.actor.shape.rot.x = var_a3->urot.x;
+            if (var_a3->action != 1) {
+                Path* var_v0_2 = &globalCtx->setupPathList[(this->dyna.actor.params >> 7) & 0x1F];
+                if (var_a3->action == 3) {
+                    var_v0_2 = &globalCtx->setupPathList[var_v0_2->unk1];
+                }
+                this->unk163 = var_v0_2->count;
+                this->unk164 = Lib_SegmentedToVirtual(var_v0_2->points);
+                Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, this->unk164);
+                this->dyna.actor.speedXZ = var_a3->urot.z * 0.005493164f;
+                this->unk164++;
+                this->unk15C = 1;
+            }
+            this->unk15F = var_a3->action;
+        } else {
+            if (var_a3->action != 1) {
+                Vec3f sp34;
+                f32 temp;
+                Math_Vec3s_ToVec3f(&sp34, this->unk164);
+                temp = Math_Vec3f_StepTo(&this->dyna.actor.world.pos, &sp34, this->dyna.actor.speedXZ);
+                if ((this->unk15C < this->unk163) && (temp < this->dyna.actor.speedXZ)) {
+                    this->unk164++;
+                    this->unk15C++;
+                }
+            }
+            if (var_a3->action != 3) {
+                func_80B9B124(this);
+                if (var_a3->action == 2) {
+                    func_800B9010(&this->dyna.actor, 0x2154U);
```

Comment:
```suggestion
                    func_800B9010(&this->dyna.actor, NA_SE_EV_PIRATE_SHIP - SFX_FLAG);
```
Can replace this with the correct SFX